File: src/main/kotlin/xingkong/loopicserver/picproc/tagImg/FilterResultStorage.java
package xingkong.loopicserver.picproc.tagImg;

import java.util.HashMap;

/*存储filter的结果*/
public class FilterResultStorage {
    HashMap<String, FilterResult> resultHash;

    public FilterResultStorage() {
        resultHash = new HashMap<String, FilterResult>();
    }

    public int searchResultNum(String key) {
        int ret = -1;
        if (resultHash.containsKey(key)) {
            FilterResult fr = resultHash.get(key);
            ret = fr.getRes();
        }
        return ret;
    }

    public FilterResult get(String key) {
        if (resultHash.containsKey(key)) {
            return resultHash.get(key);
        }
        return null;
    }

    public void put(String key, FilterResult value) {
        resultHash.put(key, value);
    }

}

File: src/main/kotlin/xingkong/loopicserver/picproc/tagImg/FilterResultStorage.kt
package xingkong.loopicserver.picproc.tagImg

import java.util.*

/*存储filter的结果*/
class FilterResultStorage {
    internal var resultHash: HashMap<String, FilterResult>

    init {
        resultHash = HashMap()
    }

    fun searchResultNum(key: String): Int {
        var ret = -1
        if (resultHash.containsKey(key)) {
            val fr = resultHash[key]
            ret = fr?.res!!
        }
        return ret
    }

    operator fun get(key: String): FilterResult? {
        return if (resultHash.containsKey(key)) {
            resultHash[key]
        } else null
    }

    fun put(key: String, value: FilterResult) {
        resultHash[key] = value
    }

}
